cmd/controller/controller.go

```diff
@@ -349,40 +320,9 @@ func updateSealedSecretsStatusConditions(st *ssv1alpha1.SealedSecretStatus, unse
 	}
 }
 
-func (c *Controller) updateSecret(ctx context.Context, newSecret *corev1.Secret) (*corev1.Secret, error) {
-	existingSecret, err := c.sclient.Secrets(newSecret.GetObjectMeta().GetNamespace()).Get(ctx, newSecret.GetObjectMeta().GetName(), metav1.GetOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("failed to read existing secret: %s", err)
-	}
-	existingSecret = existingSecret.DeepCopy()
-	existingSecret.Data = newSecret.Data
-
-	c.updateOwnerReferences(existingSecret, newSecret)
-
-	return existingSecret, nil
-}
-
-func (c *Controller) updateOwnerReferences(existing, new *corev1.Secret) {
-	ownerRefs := existing.GetOwnerReferences()
-
-	for _, newRef := range new.GetOwnerReferences() {
-		found := false
-		for _, ref := range ownerRefs {
-			if newRef.UID == ref.UID {
-				found = true
-				break
-			}
-		}
-		if !found {
-			ownerRefs = append(ownerRefs, newRef)
-		}
-	}
-	existing.SetOwnerReferences(ownerRefs)
-}
```

Comment:
@mkmik do you know why is this unused? Tech debt?

Reply:
it's a leftover from some refactoring, possibly when the k8s client API got updated.

it's dead code, let's remove it